### ISSUE TITLE
Add MasterDetail Grid Option 

### ISSUE THREAD

**Issue by wvitpr** [Servoy/aggridcomponents#30]
Would it be possible to add MasterDetail option to AgGrid?..

**Comment by seanthomasdevlin** [comment]
Hi wvitpr,

This is already possible using NG Grids with other Servoy components. Can you explain your use case with more detail?

Thanks,
Sean

**Comment by wvitpr** [comment]
Hi Sean,
What I'm trying to achieve is a grid within a grid. So I single row expanding to a grid of multiple rows, I guess like a treetable view, but unfortunately its not available at the moment for NG Client, as it was in my pre upgraded 7.3 solution, AG grid  also has a treedata view that would be great if implemented.

**Comment by seanthomasdevlin** [comment]
Hi wvitpr,

The grouping is not sufficient for this use case ? expanding a row to get more rows?
Or the columns will be different? How would that even look ?

**Comment by Winfly-NG** [comment]
Hi Sean,

This feature would be very useful, for example the possibility to expand the order to show the order details.
Is it something already possibile in servoy data grid?

Thanks,
Andrea

**Comment by gboros** [comment]
I guess something like this? https://www.ag-grid.com/javascript-grid-master-detail-custom-detail/ ? But then how would you provide the info for the detail? just providing some html formated data via a dataprovider? as rendering a form insided a table don't seem possible ...

**Comment by wvitpr** [comment]
Hi Sean,

Could we not just have a nested grid of related records ? so nggrid within
nggrid. ?



*From:* Gabi <notifications@github.com>
*Sent:* Monday, 20 July 2020 5:05 PM
*To:* Servoy/aggridcomponents <aggridcomponents@noreply.github.com>
*Cc:* wvitpr <proutley@wvit.com.au>; State change <
state_change@noreply.github.com>
*Subject:* Re: [Servoy/aggridcomponents] Add MasterDetail Grid Option (#30)



I guess something like this?
https://www.ag-grid.com/javascript-grid-master-detail-custom-detail/ ? But
then how would you provide the info for the detail? just providing some
html formated data via a dataprovider? as rendering a form insided a table
don't seem possible ...

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/Servoy/aggridcomponents/issues/30#issuecomment-660845102>,
or unsubscribe

**Comment by gboros** [comment]
showing a grid is also possible, https://www.ag-grid.com/javascript-grid-master-detail-detail-grids/ , then also, the question is, how would you feed the data to the inside grid?

**Comment by wvitpr** [comment]
Via a relationship or dataset

**Comment by Winfly-NG** [comment]
I agree, the best way would be by relationship or dataset in my opinion

**Comment by gboros** [comment]
ok, so having a detail view with a table is what we want (and not a form), then we can have 2 properties: detailViewFoundset (dataset for powergrid) and detailViewColumns

**Comment by gboros** [comment]
created a case on Servoy support https://support.servoy.com/browse/SVY-15231

**Comment by seanthomasdevlin** [comment]
To have a real detail form (not a grid in a grid) would be very challenging. If that is required, I recommend using the collapse component, which can be built programmatically. It's not a grid in the header but can be injected with markup for formatting.
https://github.com/Servoy/servoy-extra-components/wiki/Collapse

**Comment by andrea-colombara-crm** [comment]
Hello, are you planning to build this feature in the future?

**Comment by seanthomasdevlin** [comment]
Hi @andrea-colombara-crm 

Please track this ticket in our Jira:
https://support.servoy.com/browse/SVY-15231

It looks like it is planned for 2026.6, but It will be a large feature to implement. Are you already on the NG Titanium version?
